src/components/fields/image-upload/types.ts

```diff
@@ -53,9 +54,11 @@ export enum EImageStatus {
 	UPLOAD_READY = 4,
 	UPLOADING = 5,
 	UPLOADED = 6,
+	PENDING = 7,
 }
 
 export interface IImage {
+	id: string;
 	file: File;
 	name: string;
 	type?: string;
```

Comment:
sorry missed this out, help me add `undefined` type too

```suggestion
	type?: string | undefined;
```